src/reducers-actions/trades.ts

```diff
@@ -161,28 +186,97 @@ function applyRevertsToTrades(
   return [flattenMapOfLists(tradesByRevertKey), getPendingTrades(tradesByRevertKey)]
 }
 
+// ******** REDUCER
+
 export const reducer = (state: TradesState, action: ReducerActionType): TradesState => {
   switch (action.type) {
     case 'APPEND_TRADES': {
-      const { trades: currTrades, pendingTrades: currPendingTrades } = state
-      const { trades: newTrades, reverts, lastCheckedBlock } = action.payload
+      const { trades: newTrades, reverts, lastCheckedBlock, networkId } = action.payload
+      const { trades: currTrades, pendingTrades: currPendingTrades } = state[networkId]
 
       const [trades, pendingTrades] = applyRevertsToTrades(newTrades, reverts, currPendingTrades)
 
-      return { trades: currTrades.concat(trades), lastCheckedBlock, pendingTrades }
+      return { ...state, [networkId]: { trades: currTrades.concat(trades), lastCheckedBlock, pendingTrades } }
     }
     case 'OVERWRITE_TRADES': {
-      const { trades: newTrades, reverts, lastCheckedBlock } = action.payload
+      const { trades: newTrades, reverts, lastCheckedBlock, networkId } = action.payload
 
       const [trades, pendingTrades] = applyRevertsToTrades(newTrades, reverts)
 
-      return { trades, lastCheckedBlock, pendingTrades }
+      return { ...state, [networkId]: { trades, lastCheckedBlock, pendingTrades } }
     }
     case 'UPDATE_BLOCK': {
-      return { ...state, ...action.payload }
+      const { networkId, lastCheckedBlock } = action.payload
+
+      return { ...state, [networkId]: { ...state[networkId], lastCheckedBlock } }
     }
     default: {
       return state
     }
   }
 }
+
+// TODO: use the one from David once his changes are merged https://github.com/gnosis/dex-react/pull/1091
+function setStorageItem(key: string, data: unknown): void {
+  // localStorage API accepts only strings
+  // TODO: consider switching to localForage API (accepts all types)
+  const formattedData = JSON.stringify(data)
+  return localStorage.setItem(key, formattedData)
+}
+
+// ******** SIDE EFFECT
+
+export async function sideEffect(state: TradesState, action: ReducerActionType): Promise<void> {
+  switch (action.type) {
+    case 'APPEND_TRADES':
+    case 'OVERWRITE_TRADES':
+    case 'UPDATE_BLOCK':
+      setStorageItem(TRADES_LOCAL_STORAGE_KEY, state)
+  }
+}
+
+// ******** INITIAL STATE / LOCAL STORAGE
+
+const INITIAL_TRADES_STATE_SINGLE_NETWORK = { trades: [], pendingTrades: new Map<string, Trade[]>() }
+
+/**
+ * Custom json parser for BN and BigNumber values.
+ * Since there's no context on what is BN/BigNumber,
+ * we have to keep track of keys and parse accordingly
+ */
+function reviver(key: string, value: unknown): unknown {
```

Comment:
there's no way of typing this more?

I'm fine with this solution if there's not. 
Other alternative would be to define a Stringify frieendly type version of the trade, where we replace the BN and BigNumbers by strings, so instead of the reviver we would have a a couple of convertertes from "Trade <--> TradeStrigifyable"

Reply:
> there's no way of typing this more?

There might be, but then I'll have to map all possible JSON serialized types to all JS deserialized types. Which feels like unnecessary to me.
Unless @Velenir has a better idea? (Of course he has)

> Other alternative would be to define a Stringify frieendly type version of the trade, where we replace the BN and BigNumbers by strings

I considered a slightly different approach: 
1. When serializing JSON, add a "watermark" to identify BN/BigNumber, such as:
    ```
    if (value is BN) return `BN|${value.toString()}`
    ```
2. On deserialization, we can have a more generic `reviver`, such as:
    ```
    if (typeof value === 'string' && /^BN|/.test(value)) return toBn(value)
    ```

I decided not to because @W3stside had a TODO for using a different type of storage:
`// TODO: consider switching to localForage API (accepts all types)`

So that might address the issue in another way.

Reply:
> there's no way of typing this more?

Don't think there is. JSON.parse, not a generic, `string` in `any` out, can only cast or assign to known type, as `TradesState` here. Same for `reviver`

Can't do a string subtype `"BN:*"`